runtime/src/main/kotlin/kotlin/Annotations.kt

```diff
@@ -148,4 +149,23 @@ public annotation class DslMarker
 @Retention(AnnotationRetention.BINARY)
 @MustBeDocumented
 @SinceKotlin("1.1")
-public annotation class PublishedApi
\ No newline at end of file
+public annotation class PublishedApi
+
+/**
+ * This annotation indicates what exceptions should be declared by a function when compiled to a platform method.
+ *
+ * When compiling to Objective-C/Swift framework, functions having or inheriting
+ * this annotation are represented as `NSError*`-producing methods in Objective-C
+ * and as `throws` methods in Swift.
+ *
+ * When Kotlin function called from Swift/Objective-C code throws an exception
+ * which is an instance of one of the [exceptionClasses] or their subclasses,
+ * it is propagated as `NSError`. Other Kotlin exceptions reaching Swift/Objective-C
+ * are considered unhandled and cause program termination.
+ *
+ * @property exceptionClasses the list of checked exception classes that may be thrown by the function.
+ */
+@SinceKotlin("1.4")
+@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER, AnnotationTarget.CONSTRUCTOR)
+@Retention(AnnotationRetention.BINARY)
+public actual annotation class Throws(actual vararg val exceptionClasses: KClass<out Throwable>)
```

Comment:
The fully qualified name has been changed in the result, but I don't think it's worth hassle to provide a migration for that.

Reply:
Can we provide `typealias Throws` in `kotlin.native` package?

Reply:
This would likely have the same ambiguity consequences as having `kotlin.Throws` and `kotlin.jvm.Throws` in default-imported packages in Kotlin/JVM. That's what I call a hassle.

Reply:
I get this.
I mean that you probably have a solution for this ambiguity for JVM. How hard would it be to apply the same solution to Native?